src/storage/redis_db.cc

```diff
@@ -309,10 +314,12 @@ rocksdb::Status Database::Keys(const std::string &prefix, std::vector<std::strin
   return rocksdb::Status::OK();
 }
 
-rocksdb::Status Database::Scan(const std::string &cursor, uint64_t limit, const std::string &prefix,
-                               std::vector<std::string> *keys, std::string *end_cursor, RedisType type) {
+rocksdb::Status Database::Scan(const std::string &cursor, uint64_t scan_success_limit, uint64_t scan_false_limit,
+                               const std::string &match_str, std::vector<std::string> *keys, std::string *end_cursor,
+                               RedisType type, const MatchType match_mode) {
```

Comment:
I have some questions about your design.

But before that, could you give a little introduction about your general design and supported patterns in the PR description?

Then we can have a detailed review on your code.

Refer to Redis glob-style patterns:
>Supported glob-style patterns:
>
>`h?llo` matches hello, hallo and hxllo
`h*llo` matches hllo and heeeello
`h[ae]llo` matches hello and hallo, but not hillo
`h[^e]llo` matches hallo, hbllo, ... but not hello
`h[a-b]llo` matches hallo and hbllo
>

Reply:
I just added two matching patterns.
such as
*llo matches ello,hello,eeeello
*l* matches l,ll,llo,ello,hello

And，i add scan_false_time limit to config,beacasue if we use suffix,substring match, we need scan one by one, and the potentially long time may be unacceptable. so I add this argument in scan,zsan,sscan,hscan command。I found that the three scan methods to get the last scanned key are all different, which is not convenient for me to modify，so i add a string value  in  CommandSubkeyScanBase to represented the last key that was scanned。

Reply:
I've been thinking about how to design a scalable scan, but I don't have any ideas. This might be a bit difficult for me. I think I need some guidance on the thought process right now.